mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -597,5 +598,16 @@ LogicalResult SYCLHostScheduleKernel::verify() {
   return verifyNdRange(*this, range, offset, ndRange);
 }
 
+LogicalResult
+SYCLHostSubmitOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  SymbolRefAttr symbol = getCGFNameAttr();
+  auto cgf =
+      symbolTable.lookupNearestSymbolFrom<LLVM::LLVMFuncOp>(*this, symbol);
+  if (!cgf)
```

Comment:
So any llvm,func is legal ?  Even if the function is not defined in this translation unit ?

Reply:
also don't understand why llvm.func is legal, and other function operation like gpu.func or func.func are illegal. we expect all device functions to be func.func or gpu.func during creation of SYCLHostSubmitOp, and CGF to be llvm.func?

Reply:
The command-group function is a function on the host, not on the device. As such, it makes sense to accept `llvm.func` (coming from `mlir-translate`)  and not any device functions.

Reply:
I added more checks here